docs/markdown/Machine-files.md

```diff
@@ -84,21 +121,76 @@ command line will override any options in the native file. For example, passing
 
 In addition to special data that may be specified in cross files, this
 section may contain random key value pairs accessed using the
-`meson.get_external_property()`
+`meson.get_external_property`, or `meson.get_cross_property`.
 
-## Properties
+*Changed in 0.55.0* putting `<lang>_args` and `<lang>_link_args` in the
+properties section has been deprecated, and should be put in the built-in
+options section.
 
-*New for native files in 0.54.0*
+### Project specific options
 
-The properties section can contain any variable you like, and is accessed via
-`meson.get_external_property`, or `meson.get_cross_property`.
+*New in 0.55.0*
+
+Path options are not allowed, those must be set in the `[paths]` section.
+
+Being able to set project specific options in a cross or native file can be
+done using the `[project options]` section of the specific file (if doing a
+cross build the options from the native file will be ignored)
+
+An incomplete list of options is:
+- pkg_config_path
+- cmake_prefix_path
+
+For setting options in supbprojects use the `<subproject>:project options`
+section instead.
+
+```ini
+[project options]
+build-tests = true
+
+[zlib:project options]
+build-tests = false
+```
+
+### Meson built-in options
+
+Meson built-in options can be set the same way:
+
+```ini
+[built-in options]
+c_std = 'c99'
+```
+
+You can set some meson built-in options on a per-subproject basis, such as
+`default_library` and `werror`. The order of precedence is:
+1) Command line
+2) Machine file
+3) build system definitions
+
+```ini
+[zlib:built-in options]
+default_library = 'static'
+werror = false
+```
+
+Options set on a per-subproject basis will inherit the
+option from the parent if the parent has a setting but the subproject
+doesn't, even when here is a default set meson language.
+
+
+```ini
+[built-in options]
+default_library = 'static'
+```
+
+will make subprojects use default_library as static.
```

Comment:
Edited for clarity:

``````suggestion
Note that the order of precedence is the following:
1) Command line
2) Machine file
3) Build system definitions

You can set some meson built-in options on a per-subproject basis, such as
`default_library` and `werror`:

```ini
[zlib:built-in options]
default_library = 'static'
werror = false
```

Per-subproject options use the same order of precedence as the root project options.
Subproject will also inherit options from the parent project if it was not specified
specifically for the subproject. For example the following:

```ini
[built-in options]
default_library = 'static'
```

will make subprojects use `default_library` as static.
``````